File: pet-vue/src/api/pet.ts
import request from '@/utils/request'

export interface Pet {
  id: number
  name: string
  breed: string
  age: string
  type: string
  gender: string
  image: string
  description: string
  healthStatus: string
  status: number
  story: string
  size: string
  personality: string
  createTime: string
}

export interface PetQuery {
  name?: string
  type?: string
  breed?: string
  age?: string
  minAge?: number
  maxAge?: number
  status?: number
  page?: number
  size?: number
}

export interface AdoptionApply {
  petId: number
  reason: string
  address: string
  phone: string
  name?: string
  email?: string
  applicantAge?: number
  housingType?: string
  hasPetExperience?: string
  familyStatus?: string
  agreeHealthCheck?: boolean
  agreeNeuter?: boolean
  agreeGoodEnvironment?: boolean
  agreeTimelyMedical?: boolean
}

export function getPetList(params: PetQuery) {
  return request.get<any, { data: { records: Pet[]; total: number } }>('/pet/list', {
    params,
  })
}

export function getPetById(id: number) {
  return request.get<any, { data: Pet }>(`/pet/${id}`)
}

export function applyAdoption(data: AdoptionApply) {
  return request.post('/adoption/apply', data)
}

export function getMyAdoptions(page = 1, size = 10) {
  return request.get<any, { data: { records: Pet[]; total: number } }>('/adoption/my', {
    params: { page, size },
  })
}

export interface AdoptionRecord {
  id: number
  userId: number
  petId: number
  status: number
  reason: string
  address: string
  phone: string
  email: string
  applicantAge: number
  housingType: string
  hasPetExperience: string
  familyStatus: string
  agreeHealthCheck: boolean
  agreeNeuter: boolean
  agreeGoodEnvironment: boolean
  agreeTimelyMedical: boolean
  createTime: string
  updateTime: string
  petName: string
  petImage: string
  petBreed: string
  petType: string
}

export function getMyAdoptionRecords(page = 1, size = 10) {
  return request.get<any, { data: { records: AdoptionRecord[]; total: number } }>('/adoption/my/records', {
    params: { page, size },
  })
}

File: pet-springboot/pet-admin/src/main/java/com/pet/admin/entity/Pet.java
package com.pet.admin.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("pet_info")
public class Pet implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;
    private String breed;
    private String age;
    private String type;
    private String gender;
    private String image;
    private String description;
    private String healthStatus;
    private Integer status;
    private Long shelterId;
    private String story;
    private String size;
    private String personality;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}
